tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -0,0 +1,109 @@
+            exit 1
+        fi
+
+        # set the value for requestResultsFile
+        echo "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt" > $(results.requestResultsFile.path)
+
+        # iib is the default internal pipeline for fbc requests
+        request=$(jq -r '.fbc.request // "iib"' ${DATA_FILE})
+
+        default_build_timeout_seconds="300"
+        iib_overwrite_from_index_credential=$(jq -r \
+          '.fbc.iibOverwriteFromIndexCredential // "iib-overwrite-fromimage-credentials"' ${DATA_FILE})
+        iib_service_config_secret=$(jq -r '.fbc.iibServiceConfigSecret // "iib-services-config"' ${DATA_FILE})
+        iib_service_account_secret=$(jq -r '.fbc.iibServiceAccountSecret // "iib-service-account"' ${DATA_FILE})
+        build_tags=$(jq -r '.fbc.buildTags // "[]"' ${DATA_FILE})
+        add_arches=$(jq -r '.fbc.addArches // "[]"' ${DATA_FILE})
+        build_timeout_seconds=$(jq -r --arg build_timeout_seconds ${default_build_timeout_seconds} \
+            '.fbc.buildTimeoutSeconds // $build_timeout_seconds' ${DATA_FILE})
+        target_index=$(jq -r '.fbc.targetIndex' ${DATA_FILE})
+        fbc_fragment=$(jq -cr '.components[0].containerImage' ${SNAPSHOT_PATH} |tr -d "\n")
+
+        echo "Creating InternalRequest to add FBC contribution to index image:"
+        internal-request -r "${request}" \
+            -p binaryImage=$(params.binaryImage) \
+            -p fromIndex=$(params.fromIndex) \
+            -p targetIndex=${target_index} \
+            -p fbcFragment=${fbc_fragment} \
+            -p iibServiceConfigSecret=${iib_service_config_secret} \
+            -p iibServiceAccountSecret=${iib_service_account_secret} \
+            -p iibOverwriteFromIndexCredential=${iib_overwrite_from_index_credential} \
+            -p buildTimeoutSeconds=${build_timeout_seconds} \
+            -p buildTags=${build_tags} \
+            -p addArches=${add_arches} \
+            -t $(params.requestTimeout) | \
+            tee $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt
+
+        internalRequest=$(awk 'NR==1{ print $2 }' $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt | xargs)
+        echo "done (${internalRequest})"
+
+        # fetching internalRequest status and populating results
+        kubectl get internalrequest ${internalRequest} -o jsonpath='{.status.results}' \
+        | tee $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt
+        conditions=$(kubectl get internalrequest ${internalRequest} \
+          -o jsonpath='{.status.conditions[?(@.type=="Succeeded")]}')
+
+        jq '.genericResult' $(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt \
+        | tee $(results.fbcOptIn.path)
+        jq '.reason // "Unset"'  <<< "${conditions}" | tee $(results.requestReason.path)
+        jq '.message // "Unset"' <<< "${conditions}" | tee $(results.requestMessage.path)
+
+        # check the InternalRequest reason only when the request was executed by
```

Comment:
So the task only fails if there are conditions and they're not successful? What if things completely blow up in this script? E.g. the internal service doesn't work and does not process the internalRequest at all. AFAICS this script would still finish successfully. Can we at least have `set -e`?

Reply:
Sure. Added.

Reply:
I will move this back to draft as this can actually be improved.

Reply:
Cool. I was advocating for the same thing on another task not long ago, but the counter argument was that we want to support async call to `internal-request`. So we don't intend to do that here? Just asking. Less code means less things that can go wrong :)

Reply:
Although I heard several times that we should go (now) for sync, I still don't like the idea of eventually having a FBC with dozen of components and serialize all operations (calling iib, signing and pushing the image), so yes, this is not the case now but I hope we change the view on this soon.

Reply:
Yeah, you're right, it makes sense that eventually we'll have to make it possible to run in parallel.